src/client/config.rs

```diff
@@ -146,15 +153,44 @@ impl Config {
     /// trust-chain.
     ///
     /// # Panics
-    /// Will panic in case `trust_cert` was called before.
+    /// Will panic in case `trust_cert` or `trust_cert_ca_pem` was called before.
     ///
     /// - Defaults to validating the server certificate is validated against system's certificate storage.
     pub fn trust_cert_ca(&mut self, path: impl ToString) {
-        if let TrustConfig::TrustAll = &self.trust {
-            panic!("'trust_cert' and 'trust_cert_ca' are mutual exclusive! Only use one.")
-        } else {
-            self.trust = TrustConfig::CaCertificateLocation(PathBuf::from(path.to_string()))
-        }
+        self.trust = match &self.trust {
+            TrustConfig::Default | TrustConfig::CaCertificateLocation(_) => {
+                TrustConfig::CaCertificateLocation(PathBuf::from(path.to_string()))
+            }
+            TrustConfig::CaCertificatePem(_) => {
+                panic!("'trust_cert_ca_pem' and 'trust_cert_ca' are mutual exclusive! Only use one.")
+            }
+            TrustConfig::TrustAll => {
+                panic!("'trust_cert' and 'trust_cert_ca' are mutual exclusive! Only use one.")
+            }
+        };
+    }
+
+    /// If set, the server certificate will be validated against the given CA certificate in
+    /// in addition to the system-truststore.
+    /// Useful when using self-signed certificates on the server without having to disable the
+    /// trust-chain.
+    ///
+    /// # Panics
+    /// Will panic in case `trust_cert` or `trust_cert_ca` was called before.
+    ///
+    /// - Defaults to validating the server certificate is validated against system's certificate storage.
+    pub fn trust_cert_ca_pem(&mut self, cert: impl ToString) {
```

Comment:
nit: maybe suffix with `_string` to clarify that we are passing a string literal containing a certificate?